### PR TITLE
fix: 11255: Discrepancy protobuf bucket bytes between Bucket and ParsedBucket if bucket index is zero

### PR DESCRIPTION
**Pull request by artemananiev**
Fix summary:

* Changed `ParsedBucket` to always write bucket index field during serialization, so assumptions in `Bucket` (e.g. header size) are always met
* Added a new unit test
* Fixed a couple other minor issues

Fixes: https://github.com/hashgraph/hedera-services/issues/11255
Signed-off-by: Artem Ananev <artem.ananev@swirldslabs.com>

### CODE REVIEW

**Review by imalygin — APPROVED** [review]

File: platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/Bucket.java [hunk 1]
```diff
@@ -241,8 +240,14 @@ public void putValue(final K key, final long value) {
                     bucketData.position(result.entryOffset());
                     bucketData.writeBytes(remainder);
                 }
+                if (bucketIndexFieldOffset > result.entryOffset()) {
+                    // It should not happen with default implementation, but if buckets are serialized
+                    // using 3rd-party tools, field order may be arbitrary, and "bucket index" field
+                    // may be after the deleted entry
+                    bucketIndexFieldOffset -= result.entrySize();
+                }
```
Comment (anthony-swirldslabs) [inline platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/Bucket.java:248]: Seems like this code is making a very particular assumption about where exactly that offset is located. I guess this should be fine as long as we only use a limited number of "third-party tools."

**Review by anthony-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/Bucket.java (hunk 1 above)
Comment (artemananiev) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/Bucket.java:248]: No particular assumptions. There used to be an assumption that "bucket index" field was always written first, but with this change the assumption is gone

Comment (anthony-swirldslabs) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/Bucket.java:248]: Sounds good.

Comment (imalygin) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/Bucket.java:248]: I'll updated it in the subsequent PR

**Pull request merged**